Update the versioning process (#1449)
owncloud/docs-server: docs/new-version-branch.md

1.  Commit the changes and push the new `x.y` branch. This makes the branch available for futher processing. DO NOT CREATE A PR!

**Step 3: Protect the new branch**

The branch has been pushed, add it to the rule set for protected version branches.

1. Go to the github settings of this repository and change the protection of the branch list so that
    the `x.y` branch just pushed before now gets included in the branch protection like the other active branches do.

**Step 4: Configure the master branch to use the new `x.y` branch**

This step is necessary to update the building process for content changes in this repo. 

1. Create a new `changes_necessary_for_x.y` branch based on latest `origin/master`.
1. In `.drone.star` set `latest_version` to `x.y` (on top in section `def main(ctx)`).
1. In `antora.yml`, check if the `version:` key is set to `next`.
1. In `site.yml` and in `antora.yml`, DO NOT adjust relevant `-version` keys.
1. Run a build by entering `npm run antora-local`. No build errors or warnings should occur.
1. Commit changes and push them. (Check the branch protection rules upfront so that the push passes.)

> 1. Create a Pull Request `Changes necessary for x.y`, see the [text suggestion](#text-suggestion-for-step-4) below. When CI is green, all is done correctly, merge the PR when approved. This merge does NOT add the version to the main building process. You can now add at any time content changes to this version.

Reviewer: Is it expected that they all start with `1`?

Author: Yes, sadly :sweat_smile: 
This is one of the downside of markdown where automatic numbering become a challenge.
But if you start a numbering always with e.g. `1.`, it will number it correctly :smiley: 

Asciidoc handles this differently and much more natural than MD

Reviewer: again what learned

**Step 5: Changes in the docs repo (main assembling repo)**

> This step finally sets the versions (branches) to be build. The new x.y version branch will be added and the oldest branch of the product will be removed from building. Note that merging the PR, the new version is live and the former version is removed. You should therefore merge _close before_ publishing the relevant product release. If this step is omitted or not timed well, the `latest` pointer from the webserver, which generates the redirect based on the latest tag of the product repo automatically, will generate a 404 instead pointing to the latest release!

Reviewer 2: ```suggestion
This step finally sets the versions (branches) to be built. The new x.y version branch will be added and the oldest branch of the product will be removed from building. Note that after merging the PR, the new version is live and the former version is removed. You should therefore merge _close before_ publishing the relevant product release. If this step is omitted or not timed well, the `latest` pointer from the webserver, which generates the redirect based on the latest tag of the product repo automatically, will generate a 404 instead pointing to the latest release!
```

1. In the [docs](https://github.com/owncloud/docs/blob/master/site.yml) repo, create a new `enable_<product>_x.y` branch based on latest `origin/master`, where `<product>` is the placeholder for the product this version is added for like `ocis`, or `desktop` etc.
1. In file `site.yml`, adjust the versions of the respective content source to the correct versions. Add the new one and remove the oldest one.
1. In file `global-attributes.yml` adjust all `-version` keys related to this repo according the new and former releases. Note that the values MUST NOT contain the trailing `@`. (The trailing @ character allows the value to be overwritten like from the corresponding `antora.yml` file which is only necessary for local building the corresponding docs-xxx repo.)
1. In file `global-attributes.yml` adjust all attributes that need an update relevant for that branch like compile timestamp etc.
1. Create a Pull Request `Enable <product> x.y`, see the [text suggestion](#text-suggestion-for-step-5) below.

**Step 6: Archive the dropped branch**

This step _mandatory_ needs the doc PR described before to be merged!

> 1. Unprotect the `x.z` branch which was dropped from the building process and rename it to `x_archived_x.z`. There are already branch protection rules for archived versions in place. Note that this step can be postponed if needed but it must not be processed before merging x.y in docs. Note that after renaming, also local building cant be done anymore without extra efforts - which is on purpose.

Reviewer: Same here?

Author: yes, yes, valid here too

## **Text Suggestion for Step 4**

The following text is a copy/paste suggestion for the PR, replace the branch numbers accordingly:

```
These are the changes necessary to finalize the creation of the x.y branch.

* The x.y branch is already pushed, prepared and is included in the branch protection rules.

* Note that this PR can be merged at any time but:
  * must be merged **before** applying content changes and
  * **before** merging the docs PR to include this version.

* Post merging this, we need to backport all relevant changes created in master to x.y

@mmattel @phil-davis
```

## **Text Suggestion for Step 5**

The following text is a copy/paste suggestion for the PR, replace the branch numbers accordingly:

```
Referencing: https://github.com/owncloud/docs-xxx/pull/yyy (Changes necessary for x.y)

> This PR enables the documetation for <product> x.y and drops version x.z

Reviewer 2: ```suggestion
This PR enables the documentation for <product> x.y and drops version x.z
```

> Currently on draft to avoid accidentially merging before x.y is finally out.

Reviewer 2: ```suggestion
Currently on draft to avoid accidentally merging before x.y is finally out.
```
